### PR TITLE
Removes support for kubernetes < 1.17 for AWS extension provider

### PR DESCRIPTION
**Pull request by dimitar-kostadinov**
**How to categorize this PR?**

/area open-source
/kind cleanup
/platform aws

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener-extension-provider-aws/issues/595

**Special notes for your reviewer**:

**Release note**:

```breaking operator
provider-aws no longer supports Shoots with Кubernetes version < 1.17.
```

### CODE REVIEW

**Review by ialidzhikov — COMMENTED** [review]
Can you check the `make check` failure?

```
Executing gofmt/goimports
Unformatted files detected:
pkg/webhook/controlplane/ensurer_test.go
make: *** [Makefile:119: check] Error 1
```

File: README.md [hunk 1]
```diff
@@ -31,10 +31,6 @@ This extension controller supports the following Kubernetes versions:
 | Kubernetes 1.19 | 1.19.0+     | [![Gardener v1.19 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.19%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.19%20AWS) |
 | Kubernetes 1.18 | 1.18.0+     | [![Gardener v1.18 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.18%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.18%20AWS) |
 | Kubernetes 1.17 | 1.17.0+     | [![Gardener v1.17 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.17%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.17%20AWS) |
-| Kubernetes 1.16 | 1.16.0+     | [![Gardener v1.16 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.16%20AWS/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.16%20AWS) |
```
Comment (ialidzhikov) [inline README.md:34]: You could update 

https://github.com/gardener/gardener-extension-provider-aws/blob/51af490da2b04a8aa0fddf638b1f95c0dc36137e/example/30-controlplaneexposure.yaml#L20

Comment (dimitar-kostadinov) [reply README.md:34]: Done

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**